pkg/apis/rsyslog/validation/rsyslog_relp.go

```diff
@@ -112,6 +121,17 @@ func validateLoggingRules(loggingRules []rsyslog.LoggingRule, fldPath *field.Pat
 	return allErrs
 }
 
+func validateProgramNames(programNames []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	validatingRegex, _ := regexp.CompilePOSIX(`[[:/]`)
+	for index, name := range programNames {
+		if validatingRegex.MatchString(name) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(index), name, ".programNames can't contain `[`, `:` or `/`"))
+		}
+	}
+	return allErrs
+}
```

Comment:
This does not follow the requirement from the validation issuue: `checks if programNames contains only printable ASCI characters`

So maybe you could do something like this: 
```suggestion
var printableCharactersRegex = regexp.MustCompile(`^[!-~]*$`) 
var invalidCharactersForProgramNameRegex = regexp.MustCompile(`[[:/]`) 

func validateProgramNames(programNames []string, fldPath *field.Path) field.ErrorList {
	allErrs := field.ErrorList{}
	for index, name := range programNames {
		if  invalidCharactersForProgramNameRegex.MatchString(name) {
			allErrs = append(allErrs, field.Invalid(fldPath.Index(index), name, ".programNames can't contain `[`, `:` or `/`"))
		}
		if !printableCharactersRegex.MatchString(name) {
            allErrs = append(allErrs, field.Invalid(fldPath.Index(index), name, ".programName can only contain printable characters")
	}
	return allErrs
}